pytorch/inference/docker/2.0/py3/cu118/Dockerfile.gpu

```diff
@@ -0,0 +1,303 @@
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.10.9
+ARG MAMBA_VERSION=22.11.1-4
+ARG OPEN_MPI_VERSION=4.1.5
+
+# Nvidia software versions
+ARG CUBLAS_VERSION=11.11.3.6
+ARG CUDNN_VERSION=8.8.0.121
+ARG NCCL_VERSION=2.16.5
+ARG NVML_VERSION=11.8.86
+
+# PyTorch Binaries and versions.
+ARG TORCH_URL=https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r2.0.0_inference/cu118_py310/torch-2.0.0%2Bcu118-cp310-cp310-manylinux_2_31_x86_64.whl
```

Comment:
This binary does not have MPI backend enabled so I'm not able to test the MPI backend issue https://github.com/pytorch/pytorch/issues/97507

``` 
pip install https://aws-pytorch-unified-cicd-binaries.s3.us-west-2.amazonaws.com/r2.0.0_inference/cu118_py310/torch-2.0.0%2Bcu118-cp310-cp310-manylinux_2_31_x86_64.whl
```
```
 python
Python 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:20:04) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> import torch.distributed as dist
>>> dist.is_mpi_available()
False
```

Reply:
thank you for result. am looking into it.

Reply:
as there would be a lot of work to remove to remove OpenMPI (it is installed). will keep at this time.